contrib/containers-common/containers-common.sh

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/bash -e
+#
+# Script used for FIXME FIXME
+#
+# Must be run from directory containing input specfile
+#
+
+die() {
+    echo "$(basename $0): $*" >&2
+    exit 1
+}
+
+branchversion() {
+    v=$(go list -m $1)
+    echo ${v##* }
+}
+
+
+SPECFILE=containers-common.spec
+if [[ ! -e $SPECFILE.in ]]; then
+    die "Please run me from the same directory as $SPECFILE.in"
+fi
+
+declare -A moduleversion
+for module in common image/v5 storage; do
+    path=github.com/containers/$module
+    v=$(branchversion $path)
+    if [[ -z "$v" ]]; then
+        die "Could not find version for module '$path'"
+    fi
+    moduleversion[$module]=$v
+done
+
+builddir=containers-common-${moduleversion[common]}
+mkdir -p $builddir
+
+sed -e "s/COMMON_BRANCH/${moduleversion[common]}/g" \
+    -e "s/IMAGE_BRANCH/${moduleversion[image/v5]}/g"  \
```

Comment:
I guess that works until v6.

I also guess that this directory is intended as a sort of example for someone (perhaps a packager?), and not intended as a well-maintained and -guaranteed tool set. If so, LGTM as soon as you remove the WIP.

Reply:
Well I would like to get the script to be a little more robust.  Currently we can only support released versions of the git repos.

Reply:
Here's [another iteration](https://github.com/containers/podman/files/8119076/containers-common.sh.txt), this one handles the `v5` issue and also handles getting the commit hash from unreleased versions. 

Unfortunately it is not working for the last three files:
```
...
Downloading: https://raw.githubusercontent.com/containers/storage/26c561f9a645/storage.conf
100% of   8.4 KiB |######################################################################################################| Elapsed Time: 0:00:00 Time:  0:00:00
Downloaded: storage.conf
Downloading: https://raw.githubusercontent.com/containers/common/v0.47.4/docs/Containerfile.5.md
Download failed:
404 Client Error: Not Found for url: https://raw.githubusercontent.com/containers/common/v0.47.4/docs/Containerfile.5.md
Downloading: https://raw.githubusercontent.com/containers/common/v0.47.4/docs/containerignore.5.md
Download failed:
404 Client Error: Not Found for url: https://raw.githubusercontent.com/containers/common/v0.47.4/docs/containerignore.5.md
Downloading: https://raw.githubusercontent.com/containers/common/v0.47.4/docs/links/.containerignore.5
Download failed:
404 Client Error: Not Found for url: https://raw.githubusercontent.com/containers/common/v0.47.4/docs/links/.containerignore.5
-   0.0 B Elapsed Time: 0:00:00
```